src/coreservices.cpp

```diff
@@ -423,6 +423,27 @@ void CoreServices::initialize(QApplication* pApp) {
 
     m_pTouchShift = std::make_unique<ControlPushButton>(ConfigKey("[Controls]", "touch_shift"));
 
+    // The following UI controls must be created here so that controllers can bind to them
+    // on startup.
+    m_uiControls.clear();
+    const std::vector<ConfigKey> uiKeys = {
+            ConfigKey("[Master]", "skin_settings"),
+            ConfigKey("[Microphone]", "show_microphone"),
+            ConfigKey(VINYL_PREF_KEY, "show_vinylcontrol"),
+            ConfigKey("[PreviewDeck]", "show_previewdeck"),
+            ConfigKey("[Library]", "show_coverart"),
+            ConfigKey("[Master]", "maximize_library"),
+            ConfigKey("[Samplers]", "show_samplers"),
+            ConfigKey("[EffectRack1]", "show"),
+            ConfigKey("[Skin]", "show_4effectunits"),
+            ConfigKey("[Master]", "show_mixer"),
+    };
+
+    for (const auto& key : uiKeys) {
+        m_uiControls.emplace_back(std::make_unique<ControlPushButton>(key));
```

Comment:
I am not sure if emplace_back() is semantically correct here, because it will internally call 
`new std::unique_ptr(std::make_unique<ControlPushButton>(key)));` calling an extra new.
```suggestion
        m_uiControls.push_back(std::move(std::make_unique<ControlPushButton>(key)));
```

Reply:
Why not `m_uiControls.emplace_back(key);` for less boilerplate?

Reply:
The previous version just invoked the move constructor of std::unique_ptr for constructing another std::unique_ptr in place. Not two nested pointers which would be impossible by the type signature.

Reply:
@daschuer emplace_back() invokes *placement new* which does not allocate again. It is correct here because it will invoke the move constructor of std::unique_ptr.

Reply:
Yes, both versions are correct and if we reserve memory before by std::vector.reserve() there is no memory allocation involved by emplace_back() or push_back(). 

The issue here is that the semantic of emplace_back designed for in-place construction taking a list of constructor arguments, while push_back is designed for moving an already existing object into the vector. Here we want to move an existing object created by std::make_unique() and using the constructor inside emplace_back() is unnecessary. 

In both cases we can explicitly use std::move() or not. I prefer to use it to document the intention.

Reply:
Interestingly both are implemented the same in libstdc++ 
```
       push_back(value_type&& __x)  { emplace_back(std::move(__x)); }
```

And there is code like: 
```
#if __cplusplus >= 201103L
         emplace_back(*__first);
#else
         push_back(*__first);
#endif
```

Conclusion: Use emplace_back whenever possible.

Reply:
Laymans overview on `emplace_back` vs `push_back`: https://www.youtube.com/watch?v=jKS9dSHkAZY

Reply:
Laymans overview on `emplace_back` vs `push_back`: https://www.youtube.com/watch?v=jKS9dSHkAZY